File: Web Survey/resources/js/recommender/voting_rules/LeastMisery.js
import AbstractVotingRule from "./AbstractVotingRule";

export default class LeastMisery extends AbstractVotingRule {

    static getRuleName() {
        return "Least Misery";
    }

    getDescription() {
        return `First finds the lowest rating for every track, then picks the best of these.
        This option is a "safe" option, in that it will pick tracks that everyone at least kind of likes.
        However, this can also lead to boring results.`
    }

    votingRule(song, songRatings) {
        //return min(song_ratings.values())
        return Math.min(...Object.values(songRatings));
    }

}

File: Experiment Processing/recommender/voting_rules/least_misery.py
from recommender.voting_rules.abstract_voting_rule import AbstractVotingRule


class LeastMisery(AbstractVotingRule):
    """
    "Least Misery" aggregation rule, returns the minimum rating for each track, which is then sorted in a descending
    order in the abstract class.
    """

    def get_name(self):
        return "Least Misery"

    def voting_rule(self, song, song_ratings):
        return min(song_ratings.values())


